src/main/scala/org/downtowndailybread/request/ClientRequest.scala

```diff
@@ -0,0 +1,141 @@
+         |                                  meta.is_valid
+         |               from canonical_id cid
+         |                           left join canonical_type ctype on cid.type = ctype.id
+         |                           left join client_attribute cattrib on cid.id = cattrib.client_id
+         |                           left join metadata meta on cattrib.metadata_id = meta.rid
+         |                           left join client_attribute_type cattribtype on cattrib.type_id = cattribtype.id
+         |               where 1 = 1
+         |                 and ctype.type = 'client'
+         |               order by cid.id, cattribtype.id, cattrib.rid desc) all_attribs
+         |where is_valid
+      """.stripMargin
+    val statement = conn.prepareStatement(sql)
+
+    val result = statement.executeQuery()
+
+    val clientAttributeTypes = (new ClientAttributeTypeRequest(conn)).getClientAttributeTypes()
+
+    val clients = createSeq(result, result => {
+      val attributeTypeName = result.getString("attribTypeId")
+      clientAttributeTypes.find(_.id.toString == attributeTypeName) match {
+        case Some(attrib) =>
+          Client(
+            UUID.fromString(result.getString("userId")),
+            Seq(ClientAttribute(attrib.tpe, s"""${result.getString("value")}""".parseJson))
+          )
+        case None => // This code SHOULD be unreachable in practice, but in theory we could see it.
+          throw new ClientAttributeTypeNotFoundException(attributeTypeName)
+      }
+    }).groupBy(r => r.id).map { case (k, v) => Client(k, v.flatMap(_.attributes)) }
+
+    clients.toSeq
+  }
+
+  def getClientById(id: UUID): Client = {
+    getAllClients().find(_.id == id) match {
+      case Some(client) => client
+      case None => throw new ClientNotFoundException(id)
+    }
+  }
+
+
+  def insertClient(attribs: Seq[AttribNameValuePair]): UUID = {
+    val clientId = insertCanonicalId(conn, Client)
+    val attributeTypes = new ClientAttributeTypeRequest(conn).getClientAttributeTypes()
+      .map { case ClientAttributeTypeInternal(id, atype) => (atype.name, id) }.toMap
+    val sql =
+      """
+        |insert into client_attribute
+        |    (id, client_id, type_id, value, metadata_id)
+        |VALUES (cast(? as uuid), cast(? as uuid), cast(? as uuid), cast(? as json), ?)
+      """.stripMargin
+
+    val ps = conn.prepareStatement(sql)
+    attribs.foreach { case AttribNameValuePair(attribName, attrib) =>
+      val attribId = insertCanonicalId(conn, ClientAttribute)
+      val metaId = insertMetadataStatement(conn, true)
+      ps.setString(1, attribId.toString)
+      ps.setString(2, clientId.toString)
+      ps.setString(3, attributeTypes(attribName).toString)
+      ps.setString(4, attrib.toString)
+      ps.setInt(5, metaId)
+      ps.addBatch()
+    }
+
+    val numResults = ps.executeBatch()
+
+    clientId
+  }
+
+  def updateClient(id: UUID, attribs: Seq[AttribNameValuePair]): UUID = {
```

Comment:
Do we want to return the UUID back for these kinds of functions? I'm concerned that it adds no real value. If you called it, then you had the UUID. Getting that back tells me nothing. Was something actually updated? Did it update successfully? Did I have a fucked UUID? It's impossible to know from this.